src/spiceypy/spiceypy.py

```diff
@@ -15048,6 +15198,27 @@ def vproj(a: ndarray, b: ndarray) -> ndarray:
     return stypes.c_vector_to_python(vout)
 
 
+@spice_error_check
+def vprojg(a: ndarray, b: ndarray) -> ndarray:
+    """
+    Find the projection of one vector onto another vector.
+    All vectors are of arbitrary dimension.
+
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/vprojg_c.html
+
+    :param a: The vector to be projected.
+    :param b: The vector onto which a is to be projected.
+    :return: The projection of a onto b.
+    """
+    ndim = len(a)
```

Comment:
I think you need to check that `ndim = len(a) = len(b)` otherwise you'll have issues with the CSPICE code as it expects that both input arrays, and the output array are of the same length.

Reply:
stylistically I've stopped doing much LPYL in spiceypy wrappers (although many asserts exists) to lean on the cspice error system more. looking at vprojg, it looks like there are no checks in cspice so an assert can be added